Task_1_1_1

andrey-dru-me1 opened the pull request.

andrey-dru-me1 committed:
Task_1_1_1 first edition.

molo4evan commented on task_1_1_1/src/main/java/BinHeap.java:
```diff
@@ -0,0 +1,59 @@
+
+public class BinHeap {
+
+    List<Integer> binHeap;
```
private

molo4evan commented on task_1_1_1/src/main/java/BinHeap.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.Collections;
+import java.util.List;
+
+public class BinHeap {
```
Нужны джавадоки на весь публичный контракт класса

molo4evan replied:
Нужны тесты (см. Junit, в пн буду рассказывать подробнее)

molo4evan replied:
> Кроме настроенной системы сборки gradle необходимо предоставить преподавателю скрипт без использования системы сборки для: компиляции, генерации документации и запуска решения

Нужен bat либо sh, в зависимости от ключей запускающий ту или иную команду
Для компиляции см. `javac`, для генерации доков - `javadoc`, для запуска программы - `java`

molo4evan commented on task_1_1_1/src/main/java/BinHeap.java:
```diff
@@ -0,0 +1,59 @@
+        siftUp(binHeap.size() - 1);
+    }
+
+    public int extractMin() {
```
Можно назвать pop, removeMin, или как-то наподобие, это больше будет подходить под стандартный нейминг коллекций

andrey-dru-me1 committed:
Fixed some issues teacher pointed at:
Method "extractMin" has been replaced with "pop";
The binHeap list variable has been made private and final.

andrey-dru-me1 committed:
Javadocs for every class and method have been added.

andrey-dru-me1 committed:
Workflow has been updated

andrey-dru-me1 committed:
Junit tests have been added;
The Main class have been reformatted: now it has 'List sort(List)' and 'void main(String[])' methods. List is getting from prompt arguments now, not from user's input.

andrey-dru-me1 committed:
Fixed some issues IDE pointed at.

andrey-dru-me1 committed:
Script for javadocs, compiling and running the program have been added.

andrey-dru-me1 committed:
build.gradle has been added

andrey-dru-me1 committed:
Jacoco-thing has been added to the build.gradle

molo4evan commented on task_1_1_1/src/test/java/MainTest.java:
```diff
@@ -0,0 +1,39 @@
+class MainTest {
+
+    @Test
+    void main() {
```
Тесты именуются `testОписаниеТеста`, и разные кейсы стоит не валить в 1 тест, а разнести по разным, либо сделать параметризованный тест

molo4evan commented on task_1_1_1/src/main/java/bash_script.bat:
```diff
@@ -0,0 +1,3 @@
+javadoc Main.java BinHeap.java
```
По идее, нужно, чтобы в зависимости от параметров скрипт позволял запускать разные стадии (сборка, запуск, генерация доков)

andrey-dru-me1 committed:
oop.yml has been changed

andrey-dru-me1 committed:
oop.yml has been changed

andrey-dru-me1 committed:
gradlew has been changed

andrey-dru-me1 committed:
build.gradle has been changed

andrey-dru-me1 committed:
gradle files have been added

andrey-dru-me1 committed:
build.gradle has been modified

andrey-dru-me1 committed:
jacocoTestReport.xml has been added

andrey-dru-me1 committed:
jacocoTestReport.xml has been added (another one)

andrey-dru-me1 committed:
jacocoTestReport.xml has been modified

andrey-dru-me1 committed:
jacocoTestReport.xml has been added (the third one)

andrey-dru-me1 committed:
Back to the roots

andrey-dru-me1 committed:
Something

andrey-dru-me1 committed:
Something

andrey-dru-me1 committed:
Something

andrey-dru-me1 committed:
jacoco path modified

andrey-dru-me1 committed:
Small syntax fixes to please reviewdog.

andrey-dru-me1 committed:
Small syntax fixes to please reviewdog (second try).

andrey-dru-me1 committed:
third try of reviewdog pleasing

andrey-dru-me1 committed:
Fourth try of reviewdog pleasing

andrey-dru-me1 committed:
The script has been parametrized.

andrey-dru-me1 committed:
Rewrote test file depending on teacher's advises;
More tests for the God Of Tests!

andrey-dru-me1 committed:
Fixed the testMain method.

andrey-dru-me1 committed:
Fixed some issues to please reviewdog

andrey-dru-me1 committed:
Fixed imports order in MainTest to please reviewdog

molo4evan commented on Task_1_1_1/src/test/java/MainTest.java:
```diff
@@ -0,0 +1,86 @@
+class MainTest {
+
+    @Test
+    void testExample() {
```
Этот набор вполне можно заменить 1 параметризованным тестом (см. @ParameterizedTest)

andrey-dru-me1 committed:
Parameterized test has been added.

andrey-dru-me1 committed:
Syntax has been modified a little to please the reviewdog

molo4evan commented on .github/workflows/oop.yml:
```diff
@@ -5,23 +5,21 @@ name: CI
 # Controls when the workflow will run
 on:
```
Заребейзи ветку на main, из-за конфликта не могу сделать fast forward merge

andrey-dru-me1 committed:
Merge branch 'main' into task-1-1-1
